sdk/schemaregistry/schema-registry/src/schemaRegistryClient.ts

```diff
@@ -101,6 +109,10 @@ export class SchemaRegistryClient implements SchemaRegistry {
     schema: SchemaDescription,
     options?: GetSchemaIdOptions
   ): Promise<SchemaId | undefined> {
+    const cached = this.schemaToIdMap.get(schema);
+    if (cached !== undefined) {
+      return cached;
+    }
     try {
       const response = await this.client.schema.queryIdByContent(
```

Comment:
Wouldn't we want to update the cache with this data? Or maybe this is already happening somehow?

Reply:
That seems like a good idea. Maybe I'm also misreading, but it appears the cache is only used if registerSchema is called? Is that right? I think it makes sense to also cache if you only ever get schema by ID/content if so.

Reply:
Great catch, I updated the cache with this response and added a test case.